File: php/features.php
<?php
declare(strict_types=1);

// AiPresentationGenerator SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class AiPresentationGeneratorFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new AiPresentationGeneratorBaseFeature();
            case "test":
                return new AiPresentationGeneratorTestFeature();
            default:
                return new AiPresentationGeneratorBaseFeature();
        }
    }
}

File: py/features.py
# AiPresentationGenerator SDK feature factory

from feature.base_feature import AiPresentationGeneratorBaseFeature
from feature.test_feature import AiPresentationGeneratorTestFeature


def _make_feature(name):
    features = {
        "base": lambda: AiPresentationGeneratorBaseFeature(),
        "test": lambda: AiPresentationGeneratorTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
